src/main/java/net/rptools/maptool/client/functions/LookupTableFunction.java

```diff
@@ -84,367 +90,521 @@ public Object childEvaluate(
     } else if ("createTable".equalsIgnoreCase(function)) {
-
-      checkTrusted(function);
-      FunctionUtil.checkNumberParam("createTable", params, 3, 4);
-      String name = params.get(0).toString();
-      String visible = params.get(1).toString();
-      String lookups = params.get(2).toString();
-      MD5Key asset = null;
-      if (params.size() > 3) {
-        asset = FunctionUtil.getAssetKeyFromString(params.get(3).toString());
-      }
-      LookupTable lookupTable = new LookupTable();
-      lookupTable.setName(name);
-      lookupTable.setVisible(FunctionUtil.getBooleanValue(visible));
-      lookupTable.setAllowLookup(FunctionUtil.getBooleanValue(lookups));
-      if (asset != null) lookupTable.setTableImage(asset);
-      MapTool.getCampaign().getLookupTableMap().put(name, lookupTable);
-      MapTool.serverCommand().updateCampaign(MapTool.getCampaign().getCampaignProperties());
-      return "";
-
+      return createTable(function, params);
     } else if ("deleteTable".equalsIgnoreCase(function)) {
-
-      checkTrusted(function);
-      FunctionUtil.checkNumberParam("deleteTable", params, 1, 1);
-      String name = params.get(0).toString();
-      LookupTable lookupTable = getMaptoolTable(name, function);
-      MapTool.getCampaign().getLookupTableMap().remove(name);
-      MapTool.serverCommand().updateCampaign(MapTool.getCampaign().getCampaignProperties());
-      return "";
-
+      return deleteTable(function, params);
     } else if ("getTableImage".equalsIgnoreCase(function)) {
-
-      checkTrusted(function);
-      FunctionUtil.checkNumberParam("getTableImage", params, 1, 1);
-      String name = params.get(0).toString();
-      LookupTable lookupTable = getMaptoolTable(name, function);
-      MD5Key img = lookupTable.getTableImage();
-      // Returning null causes an NPE when output is dumped to chat.
-      return Objects.requireNonNullElse(img, "");
-
+      return getTableImage(function, params);
     } else if ("setTableImage".equalsIgnoreCase(function)) {
+      return setTableImage(function, params);
+    } else if ("copyTable".equalsIgnoreCase(function)) {
+      return copyTable(function, params);
+    } else if ("setTableEntry".equalsIgnoreCase(function)) {
+      return setTableEntry(function, params);
+    } else if ("getTableEntry".equalsIgnoreCase(function)) {
+      return getTableEntry(function, params);
+    } else if ("resetTablePicks".equalsIgnoreCase(function)) {
+      return resetTablePicks(function, params);
+    } else if ("setTablePickOnce".equalsIgnoreCase(function)) {
+      return setTablePickOnce(function, params);
+    } else if ("getTablePickOnce".equalsIgnoreCase(function)) {
+      return getTablePickOnce(function, params);
+    } else if ("getTablePicksLeft".equalsIgnoreCase(function)) {
+      return getTablePicksLeft(function, params);
+    } else { // if tbl, table, tblImage or tableImage
+      return tbl_and_table(function, params);
+    }
+  }
```

Comment:
It's clearer if you convert the function variable to lower (or upper) case and then just use a switch statement, or better yet a switch expression if you are just going to return the value of a function in each else if.

Reply:
Agreed.  I'm surprised IntelliJ didn't suggest that.